### PR TITLE
Update indel realignment subworkflow

### LINKED ISSUE #147: Update Indel Realignment Subworkflow
Update the Indel Realignment Subworkflow to use ABRA2 2.22 and GATK FixMateInformation 4.1.8.0.  Make sure to port --no-edge-ci option from ABRA2 2.22

- [x] Update CWL
- [x] Update README
- [x] Example YAML

### PR DESCRIPTION
**Pull request by murphycj2**
I tested the workflow successfully using `mskaccess/abra2:0.6.3` docker image (which I believe is version 2.17). Abra 2.22 was not available for testing (i.e. `mskaccess/abra:2.22` image did not exist). The log from running the test is below.

Also, maybe it's just because I am new to cwl/toil, but i needed to make some fixes to some of the cwl tools in order to test the subworkflow (see [PR](https://github.com/msk-access/cwl-commandlinetools/pull/80)). Let me know if I misunderstood anything.

Resolves #147 

<details>
<summary>Log from running</summary>
  
```
toil-cwl-runner --maxMemory 40G --cleanWorkDir onSuccess --disableCaching 'pwd' indel_realignment.cwl example_inputs.yaml
LSKI2956 2020-09-15 16:00:07,457 MainThread WARNING toil.batchSystems.singleMachine: Not enough memory! User limited to 42949672960 bytes but we only have 34359738368 bytes.
LSKI2956 2020-09-15 16:00:07,478 MainThread INFO toil: Running Toil version 4.1.0-5ad5e77d98e1456b4f70f5b00e688a43cdce2ebe.
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmply0xs9uk/worker_log.txt
LSKI2956 2020-09-15 16:00:08,300 MainThread INFO toil.leader: 0 jobs are running, 0 jobs are issued and waiting to run
LSKI2956 2020-09-15 16:00:08,301 MainThread INFO toil.leader: Issued job 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_genomecov_v2.28.0_cv2/bedtools_genomecov_v2.28.0_cv2.cwl#bedtools_genomecov' bedtools genomecov kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_genomecov_v2.28.0_cv2_bedtools_genomecov_v2.28.0_cv2.cwl_bedtools_genomecov/instance-_8_5bq9h with job batch system ID: 1 and cores: 1, disk: 2.0 G, and memory: 19.5 G
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmpm4l0w4w5/worker_log.txt
LSKI2956 2020-09-15 16:00:27,552 MainThread INFO toil.leader: Job ended: 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_genomecov_v2.28.0_cv2/bedtools_genomecov_v2.28.0_cv2.cwl#bedtools_genomecov' bedtools genomecov kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_genomecov_v2.28.0_cv2_bedtools_genomecov_v2.28.0_cv2.cwl_bedtools_genomecov/instance-_8_5bq9h
LSKI2956 2020-09-15 16:00:27,554 MainThread INFO toil.leader: Issued job 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_merge_v2.28.0_cv2/bedtools_merge_v2.28.0_cv2.cwl#bedtools_merge' bedtools merge kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_merge_v2.28.0_cv2_bedtools_merge_v2.28.0_cv2.cwl_bedtools_merge/instance-oshqz_5d with job batch system ID: 2 and cores: 1, disk: 2.0 G, and memory: 19.5 G
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmpl2dwk8zp/worker_log.txt
LSKI2956 2020-09-15 16:00:29,481 MainThread INFO toil.leader: Job ended: 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_merge_v2.28.0_cv2/bedtools_merge_v2.28.0_cv2.cwl#bedtools_merge' bedtools merge kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_merge_v2.28.0_cv2_bedtools_merge_v2.28.0_cv2.cwl_bedtools_merge/instance-oshqz_5d
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmp76w4t10f/worker_log.txt
LSKI2956 2020-09-15 16:00:30,208 MainThread INFO toil.leader: Issued job 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/abra2_2.22/abra2_2.22.cwl#abra2_2_22' java kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_abra2_2.22_abra2_2.22.cwl_abra2_2_22/instance-v2vn2yvh with job batch system ID: 4 and cores: 4, disk: 2.0 G, and memory: 16.6 G
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmpt3z_4x28/worker_log.txt
LSKI2956 2020-09-15 16:02:22,027 MainThread INFO toil.leader: Job ended: 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/abra2_2.22/abra2_2.22.cwl#abra2_2_22' java kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_abra2_2.22_abra2_2.22.cwl_abra2_2_22/instance-v2vn2yvh
LSKI2956 2020-09-15 16:02:22,034 MainThread INFO toil.leader: Issued job 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/picard_fix_mate_information_4.1.8.1/picard_fix_mate_information_4.1.8.1.cwl#picard_fix_mate_information_4_1_8_1' java kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_picard_fix_mate_information_4.1.8.1_picard_fix_mate_information_4.1.8.1.cwl_picard_fix_mate_information_4_1_8_1/instance-mn5co9uy with job batch system ID: 5 and cores: 2, disk: 2.0 G, and memory: 24.4 G
LSKI2956 2020-09-15 16:02:22,195 Thread-15 WARNING toil.statsAndLogging: Got message from job at time 09-15-2020 16:02:22: Job used more disk than requested. Please reconsider modifying the user script to avoid the chance  of failure due to incorrectly requested resources. Job 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/abra2_2.22/abra2_2.22.cwl#abra2_2_22' java kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_abra2_2.22_abra2_2.22.cwl_abra2_2_22/instance-v2vn2yvh used 148.06% (3.0 GB [3179536384B] used, 2.0 GB [2147483648B] requested) at the end of its run.
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmpsa58gp01/worker_log.txt
LSKI2956 2020-09-15 16:02:29,698 MainThread INFO toil.leader: Job ended: 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/picard_fix_mate_information_4.1.8.1/picard_fix_mate_information_4.1.8.1.cwl#picard_fix_mate_information_4_1_8_1' java kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_picard_fix_mate_information_4.1.8.1_picard_fix_mate_information_4.1.8.1.cwl_picard_fix_mate_information_4_1_8_1/instance-mn5co9uy
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmpu907u5__/worker_log.txt
LSKI2956 2020-09-15 16:02:30,287 MainThread INFO toil.leader: Issued job 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_merge_v2.28.0_cv2/bedtools_merge_v2.28.0_cv2.cwl#bedtools_merge' bedtools merge kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_merge_v2.28.0_cv2_bedtools_merge_v2.28.0_cv2.cwl_bedtools_merge/instance-oshqz_5d with job batch system ID: 7 and cores: 1, disk: 2.0 G, and memory: 19.5 G
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmp5fl212b4/worker_log.txt
LSKI2956 2020-09-15 16:02:30,865 MainThread INFO toil.leader: Job ended: 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_merge_v2.28.0_cv2/bedtools_merge_v2.28.0_cv2.cwl#bedtools_merge' bedtools merge kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_merge_v2.28.0_cv2_bedtools_merge_v2.28.0_cv2.cwl_bedtools_merge/instance-oshqz_5d
LSKI2956 2020-09-15 16:02:30,867 MainThread INFO toil.leader: Issued job 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_genomecov_v2.28.0_cv2/bedtools_genomecov_v2.28.0_cv2.cwl#bedtools_genomecov' bedtools genomecov kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_genomecov_v2.28.0_cv2_bedtools_genomecov_v2.28.0_cv2.cwl_bedtools_genomecov/instance-_8_5bq9h with job batch system ID: 8 and cores: 1, disk: 2.0 G, and memory: 19.5 G
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmp0o_ou673/worker_log.txt
LSKI2956 2020-09-15 16:02:31,466 MainThread INFO toil.leader: Job ended: 'file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/command_line_tools/bedtools_genomecov_v2.28.0_cv2/bedtools_genomecov_v2.28.0_cv2.cwl#bedtools_genomecov' bedtools genomecov kind-file_Users_charlesmurphy_Desktop_GitHub_cwl_subworkflows_command_line_tools_bedtools_genomecov_v2.28.0_cv2_bedtools_genomecov_v2.28.0_cv2.cwl_bedtools_genomecov/instance-_8_5bq9h
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmp36n7zb28/worker_log.txt
INFO:toil.worker:Redirecting logging to /var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/node-141472b2-04da-480e-8933-c3a63b60c17b-190070690681122/tmp15bqw2jw/worker_log.txt
LSKI2956 2020-09-15 16:02:34,289 MainThread INFO toil.leader: Finished toil run successfully.

Workflow Progress 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 11/11 (0 failures) [02:25<00:00, 0.08 jobs/s]
{
    "indel_realignment_bam": {
        "location": "file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/indel_realignment/test-abra_fm.bam",
        "basename": "test-abra_fm.bam",
        "nameroot": "test-abra_fm",
        "nameext": ".bam",
        "class": "File",
        "checksum": "sha1$bf653bd52bdf1a976508c0602a0928eff03cef5e",
        "size": 12796915,
        "secondaryFiles": [
            {
                "basename": "test-abra_fm.bai",
                "location": "file:///Users/charlesmurphy/Desktop/GitHub/cwl_subworkflows/indel_realignment/test-abra_fm.bai",
                "class": "File",
                "nameroot": "test-abra_fm",
                "nameext": ".bai",
                "checksum": "sha1$7e6437bc268f8e6379f48e680016fede9fcffa9d",
                "size": 1943552
            }
        ]
    }
}LSKI2956 2020-09-15 16:02:34,342 MainThread INFO toil.common: Successfully deleted the job store: FileJobStore(/var/folders/gq/wmv8s1w165nc4nwgh_d959300000gp/T/tmpcufqoyfc)
```
</details>

<details>
<summary>The example inputs file I used.</summary>
  
```
input_bam:
  class: File
  path: "test.bam"
number_of_threads: null
reference_fasta:
  class: File
  path: "/Users/charlesmurphy/Desktop/data/ref/hg19/Homo_sapiens_assembly19.fasta"
  secondaryFiles:
    - class: File
      path: "/Users/charlesmurphy/Desktop/data/ref/hg19/Homo_sapiens_assembly19.fasta.fai"
soft_clip_contig: null
maximum_mixmatch_rate: null
maximum_average_depth: null
consensus_sequence: null
scoring_gap_alignments: null
contig_anchor: null
window_size: null
ignore_bad_assembly: null
bam_index: true
distance_between_features: null
output_bams:
  - "test-abra.bam"
no_edge_complex_indel: null
no_sort: null
option_bedgraph: true

</details>

### CODE REVIEW

**Comment by murphycj2** [comment]
also, do you want a readme file within the `indel_realignment` folder to replace the `indel_realignment.md` file (i.e. the same as we do things in `cwl-commandlinetools`)? Or do you want to keep it the way it is and just update gitbooks with the new tools?

**Comment by rhshah** [comment]
> * bedtools merge tool hooked up to anything. What is the role of that tool here?

`bedtools coverage` and `bedtools merge` were done to get coverage across all the intervals and then generate a bed file. 

> * there is an output called `indel_realign_targets`, but it too does not seemed to be hooked up to anything. Should I exclude it as well?

It would be the output for the bedtools merge. the links might have broken because of the restructuring of the repo.

**Comment by rhshah** [comment]
> also, do you want a readme file within the `indel_realignment` folder to replace the `indel_realignment.md` file (i.e. the same as we do things in `cwl-commandlinetools`)? Or do you want to keep it the way it is and just update gitbooks with the new tools?

You can follow the same thing that I did for separate bams workflow README. in the `fgbio_separate_bams` dir

File: indel_realignment/indel_realignment.cwl [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  - id: option_bedgraph
+    type: boolean?
+    'sbg:x': 14.788423538208008
+    'sbg:y': 104.05289459228516
+  - id: no_edge_complex_indel
+    type: boolean?
+    'sbg:x': 7.0559563636779785
+    'sbg:y': 1133.9771728515625
+outputs:
+  - id: abra_fx_bam
```
Comment (rhshah) [inline indel_realignment/indel_realignment.cwl:76]: Can we name this as indel_realignment_bam?

File: indel_realignment/indel_realignment.cwl [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  - id: abra2_2_22
+    in:
+      - id: number_of_threads
+        source: number_of_threads
+      - id: input_bam
+        source:
+          - input_bam
+      - id: reference_fasta
+        source: reference_fasta
+      - id: targets
```
Comment (rhshah) [inline indel_realignment/indel_realignment.cwl:92]: Shouldn't the targets come from `bedtools_merge`

Comment (murphycj2) [reply indel_realignment/indel_realignment.cwl:92]: yeah i've already made the changes. the pr is not ready to review yet. i'll let you know

**Review by rhshah — COMMENTED** [review]
Sorry, I think I might be a little early in reviewing this, I remember we speaking on Friday and I think you might still be working on incorporating those changes.

**Comment by murphycj2** [comment]
ok the PR is ready for review. I updated the PR description.

**Comment by rhshah** [comment]
> (i.e. `mskaccess/abra:2.22` image did not exist).

I just added this image let me know if it works.

**Comment by rhshah** [comment]
@murphycj2 once you confirm about [80](https://github.com/msk-access/cwl-commandlinetools/pull/80) we can merge that , then update the command line tools here and then test it. I also made some changes here feel free to review

**Comment by murphycj2** [comment]
ok i updated with most recent command line tool fixes. I also successfully tested with updated docker image

**Review by rhshah — APPROVED** [review]

**Pull request merged**